doc/tutorials/biicode/tutorial_biicode/tutorial_biicode.markdown

```diff
@@ -0,0 +1,121 @@
+
+---------
+Prerequsisites
+----------------
+This tutorial assumes that you have de following avaliable in your computer:
+
+ 1. Last version of biicode tool.
+ 2. If in Windows: Any Visual Studio version (Visual Studio 12 preferred).
+
+Installation
+------------
+The installation of biicode is quite simple, follow the next [link to install it at any OS](http://www.biicode.com/downloads).
+
+----------
+
+Example
+-------
+In this example we'll show how to detect faces in images using the [*objdetect module* from OpenCV](http://docs.opencv.org/doc/tutorials/objdetect/table_of_content_objdetect/table_of_content_objdetect.html).
+
+After donwloading and isntalling biicode, do the following steps:
+
+	$ bii init mycvproject
+	$ cd mycvproject
+	$ bii open diego/opencvex
+If in windows:
+		
+		$ bii cpp:configure -G "visual Studio 12"
+
+Then we can build the project. **Note** that this can take a while, until it downloads and builds OpenCV. However, this is done only once in your machine, in your "user/.biicode" folder. If the opencv installation process fails, you might simply want to go there, delete opencv files inside "user/.biicode" and repeat.
+
+	$ bii cpp:build
+
+Your binaries are located inside the bin folder:
+	
+	$ cd bin
+	$ ./diego_opencvex_main
+
+![](images/biiapp.png)
+
+
+	$ ./diego_opencvex_mainfaces
+
+![](images/lena.png)
+
+--------
+
+
+Developing your own application
+---------------------------------------
+
+biicode works with **#include** headers in your sourcecode files, it reads them and retrieve all the dependencies on our database. So it is as simple as typing #include *"diego/opencv/opencv/cv.h"* in the headers of your *.cpp* file.
+
+To start a new project using OpenCV, just do the following:
+
+	$ bii init mycvproject
+	$ cd mycvproject
+
+The next line just create a *myuser/myblock* folder inside "blocks" with a simple "Hello World" *main.cpp* inside it. You can also do it by hand:
+
+	$ bii new myuser/myblock --hello=cpp
+
+Now replace your *main.cpp* contents inside *blocks/myuser/myblock* with **your app code**.
+Put the includes as:
+	
+`#include "diego/opencv/opencv/cv.h`
+
+If you type:
+
+	$ bii deps
+You will check that ***opencv/cv.h*** is an "unresolved" dependency. You can find it with:
+
+	$ bii find
+
+Now, you can just configure and build your project as described above.
+
+If you want simpler *#include* directives, you can configure it in your **biicode.conf** file. Let your includes be:
+
+`#include "opencv/cv.h"`
+
+And then just write in your **biicode.conf**:
+
+	[includes]
+	     opencv/cv.h: diego/opencv
+	[requirements]
+	     diego/opencv: 0
+
+----------
+Switching OpenCV versions
+--------------------------------
+If you want to try or develop your application against **opencv 2.4.10** and also try **3.0-beta**, you can change easily in your **biicode.conf** file, simply alternating track in your requirements:
```

Comment:
and also against **3.0-beta**, you can change it easily in your **biicode.conf** file, simply alternating the track in your [requirements]: